ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule/Shrink.hs

```diff
@@ -35,10 +40,26 @@ isReasonablePointSchedule blockTree PointSchedule{ticks} =
     & filter (\Tick{active=Peer{name}} -> name == HonestPeer)
     & honestNodeIsAlert blockTree
 
--- | Shrink a 'PointSchedule' in a generic way; only yield “reasonable” point
--- schedules. See 'isReasonablePointSchedule'.
-shrinkPointSchedule :: BlockTree TestBlock -> PointSchedule -> [PointSchedule]
-shrinkPointSchedule blockTree PointSchedule{ticks, peerIds} =
+-- | Shrink an alert 'PointSchedule' in a generic way; we call a 'PointSchedule'
+-- “alert” when it contains an honest peer that behaves in an alert way. This
+-- shrinking function always yield alert 'PointSchedules'.
+shrinkAlertPointSchedule :: BlockTree TestBlock -> PointSchedule -> [PointSchedule]
+shrinkAlertPointSchedule blockTree PointSchedule{ticks, peerIds} =
   shrinkList (const []) (toList ticks)
     & mapMaybe (flip pointSchedule peerIds)
     & filter (isReasonablePointSchedule blockTree)
+
+-- | Shrink a 'PointSchedule' in a smart way: this function assumes nothing on
+-- the given 'PointSchedule' but it also gets a 'StateView' which allows it to
+-- inspect the run that this 'PointSchedule' triggered. This can help it take
+-- smart decisions of shrunk 'PointSchedule's which we know have big chances to
+-- trigger the same error.
+smartShrinkPointSchedule :: BlockTree TestBlock -> PointSchedule -> StateView -> [PointSchedule]
+smartShrinkPointSchedule _bt _ps _sv =
+  []
```

Comment:
It would be nice if these functions were documented to explain the actual shrinking strategies that they implement. Saying that the shrinking is smart or generic doesn't offer much insight currently.